firewall.go

```diff
@@ -139,3 +667,67 @@ func FirewallRuleExistsByName(rules *ole.IDispatch, name string) (bool, error) {
 
 	return false, nil
 }
+
+// firewallRulesEnum takes fwPolicy object and returns all objects which needs freeing and enum itself,
+// which is used to enumerate rules. do not forget to:
+//   defer firewallRulesEnumRealease(ur, ep)
+func firewallRulesEnum(fwPolicy *ole.IDispatch) (*ole.VARIANT, *ole.VARIANT, *ole.IEnumVARIANT, error) {
+	unknownRules, err := oleutil.GetProperty(fwPolicy, "Rules")
+	if err != nil {
+		return nil, unknownRules, nil, fmt.Errorf("failed to get Rules: %s", err)
+	}
+	rules := unknownRules.ToIDispatch()
+
+	enumProperty, err := rules.GetProperty("_NewEnum")
+	if err != nil {
+		unknownRules.Clear()
+		return nil, unknownRules, nil, fmt.Errorf("failed to get enumeration property on Rules: %s", err)
+	}
+
+	enum, err := enumProperty.ToIUnknown().IEnumVARIANT(ole.IID_IEnumVariant)
+	if err != nil {
+		enumProperty.Clear()
+		unknownRules.Clear()
+		return nil, unknownRules, nil, fmt.Errorf("failed to cast enum to correct type: %s", err)
+	}
+	if enum == nil {
+		enumProperty.Clear()
+		unknownRules.Clear()
+		return nil, unknownRules, nil, fmt.Errorf("can't get IEnumVARIANT, enum is nil")
+	}
+	return unknownRules, enumProperty, enum, nil
+}
+
+// firewallRuleEnumRelease will free memory used by firewallRulesEnum.
+func firewallRulesEnumRealease(unknownRules, enumProperty *ole.VARIANT) {
+	enumProperty.Clear()
+	unknownRules.Clear()
+}
+
+// apiInit initialize common fw api.
+// then:
+// dispatch apiRelease(u, fwp)
+func apiInit() (*ole.IUnknown, *ole.IDispatch, error) {
```

Comment:
We are in a global namespace here with the rest of the library, so do you think this function name should include "firewall", to help make sure it doesn't get confusing?